test/test_dataloader.py

```diff
@@ -752,6 +756,17 @@ def setUp(self):
         self.data = torch.randn(100, 2, 3, 5)
         self.labels = torch.randperm(50).repeat(2)
         self.dataset = TensorDataset(self.data, self.labels)
+        self.persistent_workers = False
+
+    def _get_data_loader(self, *args, **kwargs):
+        if (self.persistent_workers or
+                ('persistent_workers' in kwargs and kwargs['persistent_workers'])):
+            if (('num_workers' in kwargs and kwargs['num_workers'] == 0)
+                    or ('num_workers' not in kwargs)):
+                self.skipTest('persistent_workers and num_workers == 0 is not compatible')
```

Comment:
I understand where this is coming from, but this skips a bunch of tests where only part of them uses num_workers=0. Could you instead turn off `persistent_worker=False` when `num_workers=0` rather than skipping?